src/v2/client/handleConfiguredFlow.js

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Errors from 'util/Errors';
+import Logger from 'util/Logger';
+import Enums from 'util/Enums';
+import { interact } from './interact';
+import {
+  getTransactionMeta,
+  getSavedTransactionMeta,
+  saveTransactionMeta,
+  clearTransactionMeta,
+} from './transactionMeta';
+
+
+// checks if "desired" remeidation is available before calling `.proceed`
+// if not available, returns original idxResponse (and logs warning)
+async function proceedIfAvailable(idxState, remediation, flow) {
+  const rem = idxState.neededToProceed.find(item => item.name === remediation);
+  if (!rem) {
+    Logger.warn('Expected remediation not found, Org may be misconfigured for this flow');
+    return idxState;
+  }
+
+  try {
+    const nextIdxState = await idxState.proceed(remediation);
+    return nextIdxState;
+  }
+  catch (err) {
+    // catches and handles `Unknown remediation` errors thrown okta-idx-js
+    if (typeof err === 'string' && err.startsWith('Unknown remediation choice')) {
+      Logger.warn(`flow [${flow}] not valid with current Org configurations`);
+      return idxState;
+    }
+    else {
+      // do not catch non-`Unknown remediation` errors here
+      throw err;
+    }
+  }
+}
+
+// attempts to "step into" a specific flow by calling `.proceed` with a specific remeidation (or calls an action)
+// the "desired" remeidation is not guaranteed to be available, depends upon Org configurations
+async function stepIntoSpecificIdxFlow(idxState, flow='') {
+  switch (flow) {
+  case Enums.LOGIN_FLOW:
+  case Enums.DEFAULT_FLOW:
+  case Enums.PROCEED_WITH_FLOW:
+    // default IDX response from interact is "Login" page/flow. Do nothing
+    return idxState;
+
+  case Enums.REGISTRATION_FLOW:
+    return await proceedIfAvailable(idxState, 'select-enroll-profile', flow);
```

Comment:
'select-enroll-profile' 'unlock-account' 

these ^ are coming from RemediationConstants so lets use them instead of plain strings

Reply:
done